test/Microsoft.SqlTools.ServiceLayer.Test.Common/TestConnectionProfileService.cs

```diff
@@ -0,0 +1,206 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using Microsoft.SqlTools.ServiceLayer.Connection.Contracts;
+using Microsoft.SqlTools.ServiceLayer.Credentials;
+using Microsoft.SqlTools.ServiceLayer.Credentials.Contracts;
+using Microsoft.SqlTools.ServiceLayer.TestDriver.Driver;
+
+namespace Microsoft.SqlTools.ServiceLayer.Test.Common
+{
+    /// <summary>
+    /// Service to get connection profiles from the configured settings
+    /// to get the credentials, test driver will be used if available otherwise the credential service will be called directly
+    /// </summary>
+    public class TestConnectionProfileService
+    {
+        public static IEnumerable<TestServerIdentity> TestServers = InitTestServerNames();
+        public static ConnectionSetting Setting = InitSetting();
+
+        public TestConnectionProfileService(ServiceTestDriver driver)
+        {
+            Driver = driver;
+        }
+
+        public TestConnectionProfileService()
+        {
+        }
+
+        private ServiceTestDriver Driver { get; set; }
+
+        private ConnectionProfile GetConnectionProfile(TestServerType serverType)
+        {
+            ConnectionProfile connectionProfile = null;
+
+            //Get the server or profile name for given type to use for database connection
+            TestServerIdentity serverIdentiry = TestServers != null ? TestServers.FirstOrDefault(x => x.ServerType == serverType) : null;
+
+            //Search for the connection info in settings.json
+            if (serverIdentiry == null)
+            {
+                //If not server name found, try to find the connection info for given type
+                connectionProfile = Setting != null && Setting.Connections != null ? Setting.Connections.FirstOrDefault(x => x.ServerType == serverType) : null;
+            }
+            else
+            {
+                //Find the connection info for specific server name or profile name
+                connectionProfile = Setting != null ? Setting.GetConnentProfile(serverIdentiry.ProfileName, serverIdentiry.ServerName) : null;
+            }
+
+            if(connectionProfile == null)
+            {
+                return new ConnectionProfile
```

Comment:
Log a warning here? It really indicates testing hasn't been set up in many ways.